main/src/cgeo/geocaching/ui/dialog/CoordinatesCalculateDialog.java

```diff
@@ -0,0 +1,1096 @@
+            buttonDone.setVisibility(View.GONE);
+        } else {
+            buttonDone.setOnClickListener(inputdone);
+        }
+
+        loadCalcState();
+        resortEquations();
+        updateResult();
+
+        /**
+         * Add Watchers
+         */
+
+        ePlainLat.addTextChangedListener(new PlainWarcher());
+        ePlainLon.addTextChangedListener(new PlainWarcher());
+
+        bLatHem.setOnClickListener(new HemisphereClickListener());
+        bLonHem.setOnClickListener(new HemisphereClickListener());
+
+        for (final EditButton button : coordButtons) {
+            button.setTextChangedListener(new TextWatcher() {
+                @Override
+                public void beforeTextChanged(final CharSequence s, final int start, final int count, final int after) {
+                    // Intentionally left empty
+                }
+
+                @Override
+                public void onTextChanged(final CharSequence s, final int start, final int before, final int count) {
+                    // Intentionally left empty
+                }
+
+                @Override
+                public void afterTextChanged(final Editable s) {
+                    resortEquations();
+                    updateResult();
+                }
+            });
+        }
+
+        return v;
+    }
+
+    // Make this dialog completely fill the screen
+    @Override
+    public void onStart() {
+        super.onStart();
+        final Dialog d = getDialog();
+        if (d != null) {
+            d.getWindow().setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
+        }
+    }
+
+    // Save the current state of the calculator such that it can be restored after screen rotation (or similar).
+    @Override
+    public void onSaveInstanceState(final Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putByteArray("calc_state", SerializationUtils.serialize(saveState()));
+    }
+
+    private void displayToast(final int message) {
+        displayToast(message, false);
+    }
+
+    private void displayToast(final int message, final boolean shortToast) {
+        final AbstractActivity activity = (AbstractActivity) getActivity();
+        if (shortToast) {
+            activity.showShortToast(activity.getString(message));
+        } else {
+            activity.showToast(activity.getString(message));
+        }
+    }
+
+    // This method has largely been lifted from the CoordinateInputDialog and is used to validate
+    // coordinate specifications prior to assigning them to the Geopoint.
+    private boolean areCurrentCoordinatesValid(final boolean signalError) {
+        // then convert text to geopoint
+        final Geopoint current;
+        final String lat = getLatResult();
+        final String lon = getLonResult();
+
+        final String[] latTokens = lat.split("[ °'\"\\.]+");
+        final String[] lonTokens = lon.split("[ °'\"\\.]+");

```

Comment:
They are both using the same delimiters but the tokens themselves are quite different.  I did actually consider creating a temporary 'delimiters' string and passing that string into the functions instead of using literals but decided against it as I thought typing in the literals was clearer, but I'll do it now.